File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let left = 0
    let right = 1
    let profit = 0
    while(right < prices.length)
    {
        finalPrice = prices[right] - prices[left]
        profit = Math.max(profit, finalPrice)
        if(prices[right] < prices[left])
        {
            left++
        }
        else
        {
            right++
        }
    }

    return profit
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        profit = 0
        right = 0
        while(right < len(prices)):
            final_price = prices[right] - prices[left]
            profit = max(profit, final_price)
            if prices[left] > prices[right]:
                left += 1
            else:
                right += 1
        return profit